File: codegen/results/v002/gemma4_31b_nothink/20260414_071053/typescript/005_unit_test_writer/solution.ts
/**
 * Calculates a bulk discount based on the quantity purchased.
 * - < 10: No discount
 * - 10 to 49: 10% discount
 * - 50+: 25% discount
 */
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

interface TestCase {
    description: string;
    price: number;
    quantity: number;
    expected: number;
}

function runTests() {
    const testCases: TestCase[] = [
        {
            description: "No discount for quantity just below 10",
            price: 100,
            quantity: 9,
            expected: 900,
        },
        {
            description: "10% discount for quantity exactly 10",
            price: 100,
            quantity: 10,
            expected: 900, // 100 * 10 * 0.9
        },
        {
            description: "10% discount for quantity just below 50",
            price: 100,
            quantity: 49,
            expected: 4410, // 100 * 49 * 0.9
        },
        {
            description: "25% discount for quantity exactly 50",
            price: 100,
            quantity: 50,
            expected: 3750, // 100 * 50 * 0.75
        },
        {
            description: "25% discount for quantity well above 50",
            price: 100,
            quantity: 100,
            expected: 7500, // 100 * 100 * 0.75
        },
    ];

    testCases.forEach((tc) => {
        const result = calculateDiscount(tc.price, tc.quantity);
        
        // Using a small epsilon for floating point comparison
        if (Math.abs(result - tc.expected) < 0.0001) {
            console.log(`PASS: ${tc.description}`);
        } else {
            console.log(`FAIL: ${tc.description} (expected: ${tc.expected}, got: ${result})`);
        }
    });
}

// Execute the test suite
runTests();

File: codegen/results/v002/gemma4_31b_nothink/20260414_071053/python/005_unit_test_writer/solution.py
def calculate_discount(price: float, quantity: int) -> float:
    """
    Calculate total price after bulk discount.
    quantity < 10: no discount
    quantity 10-49: 10% off
    quantity >= 50: 25% off
    Returns: price * quantity * discount_multiplier
    """
    if quantity < 10:
        return price * quantity
    elif quantity < 50:
        return price * quantity * 0.9
    else:
        return price * quantity * 0.75


def run_tests():
    # Define test cases as (price, quantity, expected_result, description)
    test_cases = [
        (100.0, 9, 900.0, "Lower boundary for no discount (9 units)"),
        (100.0, 10, 900.0, "Lower boundary for 10% discount (10 units)"),
        (100.0, 49, 4410.0, "Upper boundary for 10% discount (49 units)"),
        (100.0, 50, 3750.0, "Lower boundary for 25% discount (50 units)"),
        (10.0, 5, 50.0, "Standard case no discount"),
        (10.0, 100, 750.0, "Large quantity 25% discount"),
    ]

    for price, quantity, expected, description in test_cases:
        actual = calculate_discount(price, quantity)
        
        # Using round() to avoid floating point precision issues during comparison
        if round(actual, 2) == round(expected, 2):
            print(f"PASS: {description}")
        else:
            print(f"FAIL: {description} (expected: {expected}, got: {actual})")

if __name__ == "__main__":
    run_tests()